src/standalone/UIKit/InputControls/CurrencyInput.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from "react";
+import { TextFieldProps } from "@material-ui/core";
+import TextFieldWithHelp from "../TextFieldWithHelp";
+import { UIInputProps } from "../CommonStyles";
+import { getGlobalized } from "../../../globalize";
+
+export interface CurrencyInputProps extends UIInputProps {
+	/**
+	 * The text for info icon
+	 */
+	infoText?: React.ReactNode;
+	/**
+	 * Boolean flag to make css important
+	 */
+	important?: boolean;
+	/**
+	 * The currency to be used in formatting
+	 */
+	currency: string;
+}
```

Comment:
missing (to be added to **all** input elements): 
```ts
value?: T
onChange?: (newValue: T) => void`;
onBlur?: React.FocusEventHandler;
``` 
T = data type of input => 
- Currency, Integer, Decimal => number | null
- Text => string
- Date, DateTime => MaterialUiPickersDate

If value != undefined the input is a controlled input => doesn't use internal state
If value == undefined the input is uncontrolled => uses internal state to keep track of it's value. internal state is of type T, formatting happens by globalize.js before passing the raw value to the TextField component

Reply:
@TQjona ,  Currency, Integer, Decimal => number | null
Here we need string type, as we are going to display formatted value in text input. We are returning number in getValue() callback.

Reply:
@ankitp-atharva How about:
```tsx
<TextField
  value={formatNumber(value)}
  onChange={parseNumber}
/>
```

Reply:
Updated as suggested